plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoProjectionPushdownPlans.java

```diff
@@ -0,0 +1,295 @@
+
+        Session session = getQueryRunner().getDefaultSession();
+
+        Optional<TableHandle> tableHandle = getTableHandle(session, completeTableName);
+        assertThat(tableHandle).as("expected the table handle to be present").isPresent();
+
+        Map<String, ColumnHandle> columns = getColumnHandles(session, completeTableName);
+
+        MongoColumnHandle column0Handle = (MongoColumnHandle) columns.get("col0");
+        MongoColumnHandle column1Handle = (MongoColumnHandle) columns.get("col1");
+
+        MongoColumnHandle columnX = createProjectedColumnHandle(column0Handle, ImmutableList.of("x"), BIGINT);
+        MongoColumnHandle columnY = createProjectedColumnHandle(column0Handle, ImmutableList.of("y"), BIGINT);
+
+        // Simple Projection pushdown
+        assertPlan(
+                "SELECT col0.x expr_x, col0.y expr_y FROM " + tableName,
+                any(tableScan(
+                        equalTo(((MongoTableHandle) tableHandle.get().getConnectorHandle()).withProjectedColumns(Set.of(columnX, columnY))),
+                        TupleDomain.all(),
+                        ImmutableMap.of("col0.x", equalTo(columnX), "col0.y", equalTo(columnY)))));
+
+        // Projection and predicate pushdown
+        assertPlan(
+                "SELECT col0.x FROM " + tableName + " WHERE col0.x = col1 + 3 and col0.y = 2",
+                anyTree(
+                        filter(
+                                "x = col1 + BIGINT '3'",
+                                tableScan(
+                                        table -> {
+                                            MongoTableHandle mongoTableHandle = (MongoTableHandle) table;
+                                            TupleDomain<ColumnHandle> constraint = mongoTableHandle.getConstraint();
+                                            return mongoTableHandle.getProjectedColumns().equals(ImmutableSet.of(column1Handle, columnX))
+                                                    && constraint.equals(TupleDomain.withColumnDomains(ImmutableMap.of(columnY, Domain.singleValue(BIGINT, 2L))));
+                                        },
+                                        TupleDomain.all(),
+                                        ImmutableMap.of("col1", equalTo(column1Handle), "x", equalTo(columnX))))));
+
+        // Projection and predicate pushdown with overlapping columns
+        assertPlan(
+                "SELECT col0, col0.y expr_y FROM " + tableName + " WHERE col0.x = 5",
+                anyTree(
+                        tableScan(
+                                table -> {
+                                    MongoTableHandle mongoTableHandle = (MongoTableHandle) table;
+                                    TupleDomain<ColumnHandle> constraint = mongoTableHandle.getConstraint();
+                                    return mongoTableHandle.getProjectedColumns().equals(ImmutableSet.of(column0Handle, columnY))
```

Comment:
We don't have to do an exact equals here - We need to ensure that - we project till the parent column.

Reply:
https://github.com/trinodb/trino/pull/17710#discussion_r1241838377